deps/rabbit/src/rabbit_queue_location.erl

```diff
@@ -63,40 +87,57 @@ leader_locator0(<<"client-local">>) ->
     <<"client-local">>;
 leader_locator0(<<"balanced">>) ->
     <<"balanced">>;
-%% 'random' and 'least-leaders' are deprecated
+%% 'random', 'least-leaders' and 'min-masters' are deprecated
 leader_locator0(<<"random">>) ->
     <<"balanced">>;
 leader_locator0(<<"least-leaders">>) ->
     <<"balanced">>;
+leader_locator0(<<"min-masters">>) ->
+    <<"balanced">>;
 leader_locator0(_) ->
     %% default
     <<"client-local">>.
 
--spec select_replicas(pos_integer(), [node(),...], [node(),...],
+-spec select_members(pos_integer(), term(), [node(),...], [node(),...],
                       non_neg_integer(), non_neg_integer(), function()) ->
     {[node(),...], function()}.
-select_replicas(Size, AllNodes, _, _, _, Fun)
+select_members(Size, _, AllNodes, _, _, _, Fun)
   when length(AllNodes) =< Size ->
     {AllNodes, Fun};
+%% Classic queues: above the threshold, pick a random node
+%% For classic queues, when there's a lot of queues, if we knew that the
+%% distribution of queues between nodes is relatively even, it'd be better
+%% to declare this queue locally rather than randomly. However, currently,
+%% counting queues on each node is relatively expensive. Users can use
+%% the client-local strategy if they know their connections are well balanced
+select_members(1, rabbit_classic_queue, _AllNodes, RunningNodes, QueueCount, QueueCountStartRandom, GetQueues)
+  when QueueCount >= QueueCountStartRandom ->
+    {RunningNodes, GetQueues};
```

Comment:
This whole function clause can be deleted given that the next function clause head matches for classic queues and returns the same value.

In general, it seems classic queue node selection can be a bit simpler than done in this PR.

Reply:
Clause removed. I also reverted one complication I introduced: since classic queues are handled by a dedicated function clause, there was no need to have classic queue support in a clause later on (when calculation `Counters`)